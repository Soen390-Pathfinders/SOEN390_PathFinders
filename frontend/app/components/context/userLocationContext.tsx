//Context to share the user location across the app for differentcomponents can access it
import React, { createContext, useContext, useState } from "react";

// Define the strcutre of the context
interface LocationContextType {
  userLocation: { latitude: number; longitude: number }; // stores the current user location
  updateUserLocation: (latitude: number, longitude: number) => void; // updates the user current location
}

// Create the context for the user location , the default value is undefined
const LocationContext = createContext<LocationContextType | undefined>(
  undefined
);

//defines LocationProvider , this needs to wrap the whole app, all the children of the  app root will have access to the user location
export const LocationProvider: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  const [userLocation, setUserLocation] = useState({
    latitude: -73.57907171854552,
    longitude: 45.49749147752672,
  }); // initial location is Hall building  (45.49749147752672, -73.57907171854552)

  // Function to update the location
  const updateUserLocation = (latitude: number, longitude: number) => {
    setUserLocation({ latitude, longitude });
  };

  return (
    <LocationContext.Provider value={{ userLocation, updateUserLocation }}>
      {children}
    </LocationContext.Provider>
  );
};

// hook to use the current user location context
export const useLocation = () => {
  const context = useContext(LocationContext);
  if (!context) {
    throw new Error("useLocation must be used within a LocationProvider");
  }
  return context;
};
export default useLocation;
