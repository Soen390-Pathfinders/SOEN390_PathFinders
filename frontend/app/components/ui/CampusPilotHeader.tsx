import { View, Text, StyleSheet } from "react-native";
import React from "react";
import AppLogo from "./AppLogo";
import { getStyles } from "@/app/styles";
import useTheme from "@/app/hooks/useTheme";
import ThemeToggle from "./ThemeToggle";
import { StatusBar } from "expo-status-bar";

export default function CampusPilotHeader() {
  const { theme, toggleTheme } = useTheme(); // Use the custom hook
  const styles = getStyles(theme); // Get styles based on theme

  return (
    /* Adding the status bar on top of the screen */
    <View style={{ flex: 0.7 }}>
      <View>
        <StatusBar style="dark" />
      </View>
      <View style={styles.iconRow}>
        <View style={styles.iconRowView}>
          <AppLogo />
        </View>
        <View>
          <Text style={[styles.logo]}>Campus Pilot</Text>
        </View>
        {/* Dark/light theme toggle */}
        <ThemeToggle />
      </View>
    </View>
  );
}
const styles = StyleSheet.create({
  headerContainer: {
    flex: 0.5, 
    paddingTop: 20, 
  },
  iconRow: {
    flexDirection: "row",
    alignItems: "center", 
    justifyContent: "space-between", 
    paddingHorizontal: 10, 
    paddingTop: 10, 
  },
  iconRowView: {
    flexDirection: "row",
    alignItems: "center", 
    gap: 10, 
  },
  logo: {
    fontSize: 20,
    fontWeight: "bold",
  },
});