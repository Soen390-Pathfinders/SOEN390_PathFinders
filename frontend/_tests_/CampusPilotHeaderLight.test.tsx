import renderer from "react-test-renderer";
import CampusPilotHeader from "@/app/components/ui/CampusPilotHeader";
import { ThemeProvider } from "@/app/components/context/ThemeContext";

jest.mock("@react-native-async-storage/async-storage", () => ({
  setItem: jest.fn(),
  getItem: jest.fn(() => Promise.resolve("light")),
  removeItem: jest.fn(),
  clear: jest.fn(),
}));

test("Headers  renders correctly in light mode", () => {
  const tree = renderer
    .create(
      <ThemeProvider>
        <CampusPilotHeader />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
