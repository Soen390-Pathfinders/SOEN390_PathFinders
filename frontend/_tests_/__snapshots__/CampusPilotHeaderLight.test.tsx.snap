// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headers  renders correctly in light mode 1`] = `
<View
  style={
    {
      "flex": 0.6,
    }
  }
>
  <View />
  <View
    style={
      {
        "alignContent": "space-around",
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "padding": 20,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <View>
        <Image
          source={
            {
              "testUri": "../../../assets/images/logo.png",
            }
          }
          style={
            {
              "height": 50,
              "width": 50,
            }
          }
          tintColor="#fff"
        />
      </View>
    </View>
    <View>
      <Text
        style={
          [
            {
              "alignItems": "center",
              "color": "#fff",
              "fontSize": 30,
              "fontWeight": "bold",
              "justifyContent": "space-around",
              "padding": 10,
            },
          ]
        }
      >
        Campus Pilot
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#000",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../assets/images/sleep-mode.png",
              }
            }
            style={
              {
                "height": 40,
                "width": 40,
              }
            }
            tintColor="#fff"
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
